### PR TITLE
Support CWL 1.1 Listing.

### PR DESCRIPTION
**Pull request by DailyDreaming**
This adds support for the new listing behavior in CWL1.1+.  See tests 242-246.

CWL1.0 tests appear to be passing: https://ucsc-ci.com/databiosphere/toil/-/jobs/53988
CWL1.1 tests appear to pass but 2 fail when caching: https://ucsc-ci.com/databiosphere/toil/-/jobs/53989

The 2 caching failures appear on the master build, so I suppose I should just make an issue to fix it since it's not directly related to this PR: https://ucsc-ci.com/databiosphere/toil/-/jobs/54084

### CODE REVIEW

File: src/toil/cwl/cwltoil.py [hunk 1]
```diff
@@ -1412,6 +1412,13 @@ def remove_unprocessed_secondary_files(unfiltered_secondary_files: dict) -> list
     return final_secondary_files
 
 
+def determine_load_listing(tool):
+    load_listing_req, _ = tool.get_requirement("LoadListingRequirement")
+    load_listing_tool_req = load_listing_req.get("loadListing", "no_listing") if load_listing_req else "no_listing"
+    load_listing = tool.tool.get("loadListing") or load_listing_tool_req
```
Comment (adamnovak) [inline src/toil/cwl/cwltoil.py:1418]: A tool is never supposed to be able to turn this off if the requirement has it set, right?

File: src/toil/cwl/cwltoil.py [hunk 2]
```diff
@@ -1412,6 +1412,13 @@ def remove_unprocessed_secondary_files(unfiltered_secondary_files: dict) -> list
     return final_secondary_files
 
 
+def determine_load_listing(tool):
```
Comment (adamnovak) [inline src/toil/cwl/cwltoil.py:1415]: There's no docstring here, so how can I tell if this code does what it's supposed to do?

File: src/toil/cwl/cwltoil.py (hunk 1 above)
Comment (DailyDreaming) [reply src/toil/cwl/cwltoil.py:1447]: Fixed!

File: src/toil/cwl/cwltoil.py (hunk 2 above)
Comment (DailyDreaming) [reply src/toil/cwl/cwltoil.py]: Fixed!

**Comment by adamnovak** [comment]
It looks like a test failed on Gitlab, but I've spent a couple minutes looking at the log and I'm unable to determine which one or how.

@DailyDreaming I'm still very unhappy with our system of dumping all the output from all the tests into the log whether they pass or fail, because when a test does fail I can't find its log in particular with ctrl+F for "fail". A lot of our tests have "fail" in their names, or in their logs, or are testing what happens when jobs fail so we get logs from failed jobs in the normal expected test output.

Can you identify which test has actually failed? And if so, how long does it take you to do it? Do you have a trick I don't know?

**Comment by DailyDreaming** [comment]
@adamnovak I usually need the full log, and I usually search for the "FAILURES =" fragment of that header : https://ucsc-ci.com/databiosphere/toil/-/jobs/54191/raw

The test structure appears in this order:

1. Setup scripts

2. All normal run output, preceded by this header:
`============================= test session starts ==============================`

3. A reiteration of all failed tests, preceded by this header:
`=================================== FAILURES ===================================`

4. Skipped tests, preceded by this header:
`=========================== short test summary info ============================`

5. Warnings, preceded by this header:
`=============================== warnings summary ===============================`

6. Test times, preceded by this header:
`============================ slowest test durations ============================`

7. Total summary:
`======= 1 failed, 48 passed, 133 skipped, 8 warnings in 1028.62 seconds ========`

If you control-f for the failures header, it should give state what failed.  I'm rerunning the tests that failed, but it looks like:


https://ucsc-ci.com/databiosphere/toil/-/jobs/54191

```
_ CachingFileStoreTestWithFileJobStore.testMultipleJobsReadSameCacheHitGlobalFile _
Traceback (most recent call last):
  File "/builds/databiosphere/toil/src/toil/test/src/fileStoreTest.py", line 632, in testMultipleJobsReadSameCacheHitGlobalFile
    self._testMultipleJobsReadGlobalFileFunction(cacheHit=True)
  File "/builds/databiosphere/toil/src/toil/test/src/fileStoreTest.py", line 666, in _testMultipleJobsReadGlobalFileFunction
    Job.Runner.startToil(A, self.options)
  File "/builds/databiosphere/toil/src/toil/job.py", line 791, in startToil
    return toil.start(job)
  File "/builds/databiosphere/toil/src/toil/common.py", line 829, in start
    return self._runMainLoop(rootJobGraph)
  File "/builds/databiosphere/toil/src/toil/common.py", line 1120, in _runMainLoop
    jobCache=self._jobCache).run()
  File "/builds/databiosphere/toil/src/toil/leader.py", line 269, in run
    raise FailedJobsException(self.config.jobStore, self.toilState.totalFailedJobs, self.jobStore)
```

and 

https://ucsc-ci.com/databiosphere/toil/-/jobs/54195

```
_ CachingFileStoreTestWithFileJobStore.testMultipleJobsReadSameCacheMissGlobalFile _
Traceback (most recent call last):
  File "/builds/databiosphere/toil/src/toil/test/src/fileStoreTest.py", line 643, in testMultipleJobsReadSameCacheMissGlobalFile
    self._testMultipleJobsReadGlobalFileFunction(cacheHit=False)
  File "/builds/databiosphere/toil/src/toil/test/src/fileStoreTest.py", line 668, in _testMultipleJobsReadGlobalFileFunction
    assert int(y.read()) > 2
AssertionError
```

**Comment by adamnovak** [comment]
@DailyDreaming Any idea what actually caused the problem with those tests? Did they fix themselves when rerun, and we have a weird intermittent bug?

**Pull request merged**